gn3/db_utils.py

```diff
@@ -24,8 +25,10 @@ def cursor(self, *args, **kwargs) -> Any:
 
 
 @contextlib.contextmanager
-def database_connection(sql_uri) -> Iterator[Connection]:
+def database_connection(sql_uri: str, logger: Optional[logging.Logger] = None) -> Iterator[Connection]:
```

Comment:
Well, if you follow my suggestion to move the creation of the logger to the module top level, you'll have to get rid of the use of `Optional` here, and line 4 above - I think that's a good (well, at the very least, acceptable) thing.

Reply:
You're right. This is a good thing. I've also dropped the `Optional` import.